### ISSUE TITLE
Java 8 compilation error version 2.5.6

### ISSUE THREAD

**Issue by PerspectivesLab** [airbnb/lottie-android#822]
lottie-react-native project target this library as 2.5.+ dependency, 
as 2.5.5 we could compile fine since it was compiled with java 1.7.
now as 2.5.6:  
Error:Error converting bytecode to dex:
Cause: Dex cannot parse version 52 byte code.
This is caused by library dependencies that have been compiled using Java 8 or above.

how to fix that ? thx

**Comment by PaulWoitaschek** [comment]
Why was this change necessary? I use desugar=false because of a different issue and that means I can't use the latest lottie version.

**Comment by PerspectivesLab** [comment]
adding 
```
   compileOptions {
        sourceCompatibility 1.8
        targetCompatibility 1.8
    }
```
requires 
```
    jackOptions {
      enabled true
    }
```
and then this error appears:
```
Lambda coming from jar file need their interfaces on the classpath to be compiled, unknown interfaces are java.util.Comparator
Message{kind=ERROR, text=Lambda coming from jar file need their interfaces on the classpath to be compiled, unknown interfaces are java.util.Comparator, sources=[E:\new\android\PerformanceTracker.java:26-27]}
```

i upgraded gradle etc to support java 8, but its a mess i broke the entire compilation
Java 8 API is min API level 24 ... which is quite high... 
what are lottie-android lines necessary in the code to be Java 8 ?

**Comment by AnujKosambi** [comment]
I temporary have solved issue by forcing gradle to take 2.5.5 version 
```
configurations.all {
    resolutionStrategy {
        force "com.airbnb.android:lottie:2.5.5"
      }
    }
```
by adding this in project's build.gradle file

**Comment by ionut-movila** [comment]
For those using Lottie in react native, the best option at the moment is described here: https://github.com/airbnb/lottie-android/issues/822#issuecomment-401776909

**Comment by WhiteHatTux** [comment]
I had to add the line from @AnujKosambi to 
`android/app/build.gradle`
In the section 
```
android{
    ...
    configurations.all {
        resolutionStrategy {
            force 'com.airbnb.android:lottie:2.5.5'
        }
    }
}
```
for it to work correctly.

**Comment by Maneeshashokan** [comment]
I have same issue, with lottie 2.5.0

error
/////////


Running dex as a separate process.

To run dex in process, the Gradle daemon needs a larger heap.
It currently has 1024 MB.
For faster builds, increase the maximum heap size for the Gradle daemon to at least 1536 MB.
To do this set org.gradle.jvmargs=-Xmx1536M in the project gradle.properties.
For more information see https://docs.gradle.org/current/userguide/build_environment.html

Uncaught translation error: com.android.dx.cf.code.SimException: invalid opcode ba (invokedynamic requires --min-sdk-version >= 26)
Uncaught translation error: com.android.dx.cf.code.SimException: invalid opcode ba (invokedynamic requires --min-sdk-version >= 26)
Uncaught translation error: com.android.dx.cf.code.SimException: invalid opcode ba (invokedynamic requires --min-sdk-version >= 26)
Uncaught translation error: com.android.dx.cf.code.SimException: invalid opcode ba (invokedynamic requires --min-sdk-version >= 26)
4 errors; aborting
:app:transformClassesWithDexForDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithDexForDebug'.
> com.android.build.api.transform.TransformException: java.lang.RuntimeException: java.lang.RuntimeException: com.android.ide.common.process.ProcessException: java.util.concurrent.ExecutionException: java.lang.UnsupportedOperationException


tried increasing heap size, no use.

**Comment by gpeal** [comment]
I'll revert the java 8 upgrade

**Comment by PerspectivesLab** [comment]
https://github.com/react-community/lottie-react-native/issues/345

**Comment by gpeal** [comment]
Fixed in https://github.com/airbnb/lottie-android/commit/17b64db1b082b11397b122f43179de7e9ddf791c

Will release 2.6.0 soon.

**Comment by cristan** [comment]
This is fixed in [2.5.7](https://github.com/airbnb/lottie-android/releases/tag/v2.5.7), but somehow 2.5.7 [never made it into the maven repository](https://mvnrepository.com/artifact/com.airbnb.android/lottie). So either we'll have to wait for 2.6.0 or 2.5.7 should be properly released.
